azurelinuxagent/common/osutil/redhat.py

```diff
@@ -140,17 +139,83 @@ def get_nm_controlled(self):
 
         return True
 
-    def publish_hostname(self, hostname):
+    def get_nic_operational_and_general_states(self, ifname):
+        """
+        Checks the contents of /sys/class/net/{ifname}/operstate and the results of 'nmcli -g general.state device show {ifname}' to determine the state of the provided interface.
+        Raises an exception if the network interface state cannot be determined.
+        """
+        filepath = "/sys/class/net/{0}/operstate".format(ifname)
+        nic_general_state_cmd = ['nmcli', '-g', 'general.state', 'device', 'show', ifname]
+        if not os.path.isfile(filepath):
+            msg = "Unable to determine primary network interface {0} state, because state file does not exist: {1}".format(ifname, filepath)
+            logger.warn(msg)
+            raise Exception(msg)
+
+        try:
+            nic_operstate = fileutil.read_file(filepath).rstrip().lower()
+            nic_general_state = shellutil.run_command(nic_general_state_cmd, log_error=True).rstrip().lower()
```

Comment:
do we need to log the error here? it's also logged at line 168 as a WARN

Reply:
removed